### DISCUSSION TITLE
Azure allocate node cidrs

### DISCUSSION THREAD

**Discussion by grunthosbird** [linkerd/linkerd2#8831] [Q&A] [answered]
Hi there,
We're trying hard to get your mesh working on our cluster with 2 nodepools - one Linux and one (shudder) windows. Now, sadly, we need this windows node. And the mesh appears to work for all the linux pods.

Looking at errors, I believe the windows pods lack the  The windows pods lack the **--allocate-node-cidrs=true** setting.
But, the most frustrating thing of all, is that I can't find out how/where to apply this setting to an Azure cluster/nodepool.

Please please help.
Obviously I'm new to all this mind chaos,
Cheers,
Dave.

**Comment by mateiidavid** [comment, marked as answer]
@grunthosbird Linkerd doesn't work with Windows nodes unfortunately. To route requests to and from the proxy in a pod's local network (i.e in the loopback interface for that specific pod), we set-up a few iptables rules. The Windows architecture doesn't support iptables; if we can't configure routing in the kernel for the host, then Linkerd will not work.

Have a look at https://github.com/linkerd/linkerd2/issues/3298 for some context.

**Reply by grunthosbird** [reply]
That's great to know - thank you.
Saves a bunch of time trying to "fix" it :)
Cheers!

**Reply by mateiidavid** [reply]
No probs. fwiw it's not a Linkerd specific problem necessarily, more of a "TCP redirection doesn't work on Windows" problem, and most service meshes would be affected by this. If you do have any suggestions on how the issue can be resolved, we'd be open to talk about it so feel free to submit an issue :)
